server/build-plugins.js

```diff
@@ -33,7 +42,78 @@ const getPackageName = name => {
   }
 };
 
-const prepareModules = async (plugins = [], local = true) => {
+async function installRemotePackages(installPackages) {
+  // check if connected to internet
+  // if not, skip npm install
+  const EXTERNAL_URI = process.env.EXTERNAL_URI || 'npmjs.com';
+  await require('dns').lookup(EXTERNAL_URI, async err => {
+    if (err && err.code === 'ENOTFOUND')
+      logger.error(`Can't reach npm servers. Skipping npm install`);
+    else {
+      // validation is done earlier, but still confirming at this step
+      const pkgStr = installPackages.reduce((str, name, index) => {
+        if (validate(name).validForNewPackages)
```

Comment:
Will `validate` ever return `null` or `undefined`? Referencing an opaque value looks a little scary

Reply:
If the api stays the same it will always have this property: https://www.npmjs.com/package/validate-npm-package-name